File: 2706-buy-two-chocolates/2706-buy-two-chocolates.cpp
class Solution {
public:
    int buyChoco(vector<int>& prices, int money) {
        
        int chocolate1 = prices[0], chocolate2 = INT_MAX;
        
        for(int i = 1; i < prices.size(); i++)
        {
            if(prices[i] >= chocolate1 and prices[i] < chocolate2)
                chocolate2 = prices[i];

            if(prices[i] <= chocolate1)
            {
                chocolate2 = chocolate1;
                chocolate1 = prices[i];
            }
        }
        int total = chocolate1 + chocolate2; 
        if(total <= money)
            return money - total;
        return money;
    }
};

File: 2706-buy-two-chocolates/2706-buy-two-chocolates.py
class Solution:
    def buyChoco(self, prices: List[int], money: int) -> int:
        
        
        choc1, choc2 = 101, 102
        
        for price in prices:
            
            if price < choc1:
                choc2 = choc1
                choc1 = price
            elif price < choc2:
                choc2 = price
        total = choc1 + choc2
        
        return money - total if total <= money else money
                
                
        
        